### PR TITLE
Authentication

### PR DESCRIPTION
**Pull request by shuklahimanshu81**
In this PR following features are added :

- User Sign Up
- User Log In
- Also options for guest login and test data for sign-up

### CODE REVIEW

File: src/backend/db/users.js [hunk 1]
```diff
@@ -12,7 +12,7 @@ export const users = [
     firstName: "Adarsh",
     lastName: "Balika",
     email: "adarshbalika@gmail.com",
-    password: "adarshbalika",
+    password: "Adarshbalika@123",
```
Comment (Jia-A) [inline src/backend/db/users.js:15]: Change these guest credentials.

**Review by Jia-A — APPROVED** [review]
Everything looks good, you can merge it.

File: src/backend/db/users.js (hunk 1 above)
Comment (shuklahimanshu81) [reply src/backend/db/users.js:15]: I didn't get you.

Comment (shuklahimanshu81) [reply src/backend/db/users.js:15]: Okay

**Review by vidhi-thakur — APPROVED** [review]
lgtm

**Pull request merged**